aws/data_source_aws_launch_configuration.go

```diff
@@ -0,0 +1,221 @@
+				Computed: true,
+			},
+
+			"enable_monitoring": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+
+			"ebs_block_device": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"delete_on_termination": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+
+						"device_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"iops": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+
+						"snapshot_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"volume_size": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+
+						"volume_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"encrypted": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"ephemeral_block_device": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"device_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"virtual_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"root_block_device": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"delete_on_termination": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+
+						"iops": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+
+						"volume_size": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+
+						"volume_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceAwsLaunchConfigurationRead(d *schema.ResourceData, meta interface{}) error {
+	autoscalingconn := meta.(*AWSClient).autoscalingconn
+	ec2conn := meta.(*AWSClient).ec2conn
+
+	if v, ok := d.GetOk("name"); ok {
+		d.SetId(v.(string))
+	}
+
+	describeOpts := autoscaling.DescribeLaunchConfigurationsInput{
+		LaunchConfigurationNames: []*string{aws.String(d.Id())},
+	}
+
+	log.Printf("[DEBUG] launch configuration describe configuration: %s", describeOpts)
+	describConfs, err := autoscalingconn.DescribeLaunchConfigurations(&describeOpts)
+	if err != nil {
+		return fmt.Errorf("Error retrieving launch configuration: %s", err)
+	}
+
+	lc := describConfs.LaunchConfigurations[0]
+
+	d.Set("key_name", lc.KeyName)
+	d.Set("image_id", lc.ImageId)
+	d.Set("instance_type", lc.InstanceType)
+	d.Set("name", lc.LaunchConfigurationName)
+	d.Set("user_data", lc.UserData)
+
+	d.Set("iam_instance_profile", lc.IamInstanceProfile)
+	d.Set("ebs_optimized", lc.EbsOptimized)
+	d.Set("spot_price", lc.SpotPrice)
+	d.Set("enable_monitoring", lc.InstanceMonitoring.Enabled)
```

Comment:
To prevent potential panics, we should `nil` check `lc.InstanceMonitoring` first and default it otherwise, e.g.

```go
d.Set("enable_monitoring", false)
if lc.InstanceMonitoring != nil {
  d.Set("enable_monitoring", lc.InstanceMonitoring.Enabled)
}
```